File: Java/1-50/17.java
import java.util.Scanner;

public class Main {
    public static int fff(int n) {
        int s = 0, p;
        while (n > 0) {
            p = n % 10;
            s = s + p;
            n = n / 10;
        }
        return s * s;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = 0;
        int i = 1, h = 0;
        while (k < n) {
            if (i % fff(i) == 0)
                k++;
            h = i;
            i++;
        }
        System.out.println(h);
    }
}

File: Kotlin/1-50/17.kt
import java.util.Scanner

fun fff(n: Int): Int {
    var s = 0
    var num = n
    while (num > 0) {
        val p = num % 10
        s += p
        num /= 10
    }
    return s * s
}

fun main() {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()
    var k = 0
    var i = 1
    var h = 0
    while (k < n) {
        if (i % fff(i) == 0) {
            k++
        }
        h = i
        i++
    }
    println(h)
}
